src/libs/actions/data-lake-alerts.ts

```diff
@@ -0,0 +1,488 @@
+    unlistenDataLakeVariable(variableId, listenerId)
+  }
+
+  delete alertStates[alertId]
+}
+
+/**
+ * Load alerts from localStorage
+ */
+const loadAlerts = (): void => {
+  try {
+    const savedAlerts = localStorage.getItem(dataLakeAlertsKey)
+    if (savedAlerts) {
+      globalAlerts = JSON.parse(savedAlerts) as DataLakeAlert[]
+      // Set up listeners for all alerts
+      globalAlerts.forEach((alert) => {
+        setupAlertListener(alert)
+      })
+    }
+  } catch (error) {
+    console.error('Failed to load data lake alerts:', error)
+    globalAlerts = []
+  }
+}
+
+/**
+ * Save alerts to localStorage
+ */
+const saveAlerts = (): void => {
+  try {
+    localStorage.setItem(dataLakeAlertsKey, JSON.stringify(globalAlerts))
+  } catch (error) {
+    console.error('Failed to save data lake alerts:', error)
+  }
+}
+
+/**
+ * Get all data lake alerts
+ * @returns {DataLakeAlert[]} All configured alerts
+ */
+export const getAllDataLakeAlerts = (): DataLakeAlert[] => {
+  return [...globalAlerts]
+}
+
+/**
+ * Get a specific alert by ID
+ * @param {string} id - The alert ID
+ * @returns {DataLakeAlert | undefined} The alert if found
+ */
+export const getDataLakeAlert = (id: string): DataLakeAlert | undefined => {
+  return globalAlerts.find((a) => a.id === id)
+}
+
+/**
+ * Create a new data lake alert
+ * @param {Omit<DataLakeAlert, 'id'>} alertConfig - The alert configuration (without ID)
+ * @returns {DataLakeAlert} The created alert
+ */
+export const createDataLakeAlert = (alertConfig: Omit<DataLakeAlert, 'id'>): DataLakeAlert => {
+  const alert: DataLakeAlert = {
+    ...alertConfig,
+    id: uuid(),
+  }
+
+  globalAlerts.push(alert)
+  setupAlertListener(alert)
+  saveAlerts()
+
+  return alert
+}
+
+/**
+ * Update an existing alert
+ * @param {DataLakeAlert} updatedAlert - The updated alert configuration
+ */
+export const updateDataLakeAlert = (updatedAlert: DataLakeAlert): void => {
+  const index = globalAlerts.findIndex((a) => a.id === updatedAlert.id)
+  if (index === -1) {
+    console.error(`Alert with id '${updatedAlert.id}' not found`)
+    return
+  }
+
+  // Clean up old listeners and set up new ones
+  cleanupAlertListener(updatedAlert.id)
+
+  globalAlerts[index] = updatedAlert
+  setupAlertListener(updatedAlert)
+
+  saveAlerts()
+}
+
+/**
+ * Delete an alert
+ * @param {string} id - The ID of the alert to delete
+ */
+export const deleteDataLakeAlert = (id: string): void => {
+  const index = globalAlerts.findIndex((a) => a.id === id)
+  if (index === -1) return
+
+  cleanupAlertListener(id)
+  globalAlerts.splice(index, 1)
+  saveAlerts()
+}
+
+/**
+ * Toggle the enabled state of an alert
+ * @param {string} id - The alert ID
+ * @param {boolean} enabled - The new enabled state
+ */
+export const setDataLakeAlertEnabled = (id: string, enabled: boolean): void => {
+  const alert = globalAlerts.find((a) => a.id === id)
+  if (!alert) return
+
+  alert.enabled = enabled
```

Comment:
As above, I think this should trigger a cleanup or setup of listeners depending on whether it is becoming disabled or enabled.